lib/platform/client/jquery_http.js

```diff
@@ -1,85 +1,101 @@
-                    var json = JSON.parse(res.responseText);
-
-                    var complete_response = that._buildResponse(error, response, json);
-                    callback(complete_response);
-                }
-            };
-            
-            return $.ajax(params);
-        },
-
-        parseJson: function(json) {
-            // JQuery does this for us
-            return json;
-        }
-    });
+
+// Copyright 2011 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"): you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+(function() {
+    var Http    = require('../../http');
+    var utils   = require('../../utils');
+    var SDK_VERSION = require('../../../package.json').version;
+
+    var root = exports || this;
+
+    var getHeaders = function(headersString) {
+        var headers = {};
+        var headerLines = headersString.split("\n");
+        for(var i = 0; i < headerLines.length; i++) {
+            if (utils.trim(headerLines[i]) !== "") {
+                var headerParts = headerLines[i].split(": ");
+                headers[headerParts[0]] = headerParts[1];
+            }
+        }
+
+        return headers;
+    };
+
+    root.JQueryHttp = Http.extend({
+        init: function(isSplunk) {
+            this._super(isSplunk);
+        },
+
+        makeRequest: function(url, message) {
+            message.headers["Splunk-Client"] = "splunk-sdk-javascript/" + SDK_VERSION;
+            if(message.response_timeout != undefined){
+                message.headers["Response-Timeout"] = message.response_timeout;
+            }
+            
+            var that = this;
+            var complete_response;
+            var params = {
+                url: url,
+                type: message.method,
+                headers: message.headers,
+                data: message.body || "",
+                timeout: message.timeout || 0,
+                dataType: "json",
+                success: utils.bind(this, function(data, error, res) {
+                    var response = {
+                        statusCode: res.status,
+                        headers: getHeaders(res.getAllResponseHeaders())
+                    };
+
+                    var complete_response = this._buildResponse(error, response, data);
+                    return Promise.resolve(complete_response)
+                }),
+                error: function(res, data, error) {
+                    // Format abort response
+                    if(res.status === 600){

```

Comment:
Can you help me understand this logic? So I see elsewhere in our code we were setting the status to 600 if the request didn't complete (no response). Is that the case being handled here? It feels like now might be a good time to fix this if so, hardcoding an invalid 600 code is probably not something we want to continue doing. I'm open to suggestions on how to address this.

Reply:
To answer your question, yes here we are handling the response status set to 600 within our code. We will think on other options for handling the 'abort' use case without hardcoding the status to 600 and let you know our findings.

Reply:
we have refactor the code such that we don't require the 600 statuscode.